nestadia/src/ppu/mod.rs

```diff
@@ -47,37 +68,47 @@ impl Ppu {
             oam_data: [0u8; 64 * 4],
             secondary_oam: [0xffu8; 8 * 4],
 
+            pattern_pipeline: [0u16; 2],
+            palette_pipeline: [0u16; 2],
+            sprites_pipeline: [0u8; 8 * 2],
+            sprites_attributes: [0u8; 8],
+            sprites_x_counter: Default::default(),
+            sprite_evaluation_state: Default::default(),
+            oam_pointer: 0,
+            secondary_oam_pointer: 0,
+            oam_latch: 0,
+            oam_temp_y_buffer: 0,
+            oam_temp_tile_buffer: 0,
+
             ctrl_reg: registers::ControlReg::default(),
             mask_reg: registers::MaskReg::default(),
             status_reg: registers::StatusReg::default(),
             oam_addr_reg: 0,
-            scroll_reg: registers::ScrollReg::default(),
-            addr_reg: registers::VramAddr::default(),
+            vram_addr: registers::VramAddr::default(),
+            temp_vram_addr: registers::VramAddr::default(),
+            fine_x: 0,
+            write_latch: false,
 
             cycle_count: 0,
-            scanline: 0,
+            scanline: -1,
             frame: [0u8; 256 * 240],
             vblank_nmi_set: false,
             last_data_on_bus: 0,
-            sprite_zero_in_secondary_oam: false,
+            sprite_zero_hit_state: Default::default(),
+            is_odd_frame: false,
+
+            nt_buffer: 0,
+            at_buffer: 0,
+            bg_lo_buffer: 0,
+            bg_hi_buffer: 0,
         }
     }
 
     pub fn reset(&mut self) {
-        self.palette_table = [0u8; 32];
-        self.oam_data = [0u8; 64 * 4];
-        self.secondary_oam = [0xffu8; 8 * 4];
-        self.ctrl_reg = registers::ControlReg::default();
-        self.mask_reg = registers::MaskReg::default();
-        self.status_reg = registers::StatusReg::default();
-        self.oam_addr_reg = 0;
-        self.scroll_reg = registers::ScrollReg::default();
-        self.addr_reg = registers::VramAddr::default();
-        self.cycle_count = 0;
-        self.scanline = 0;
-        self.frame = [0u8; FRAME_WIDTH * FRAME_HEIGHT];
-        self.vblank_nmi_set = false;
-        self.sprite_zero_in_secondary_oam = false;
+        *self = Self {
+            scanline: -1,
+            ..Self::default()
+        };
```

Comment:
Default value already sets scanline to `-1`
```suggestion
        *self = Self::default();
```

Reply:
I didn't see there was a default implementation!